src/covariances.jl

```diff
@@ -0,0 +1,76 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENCE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    designmatrix(comps)
+    
+Converts a vector of Composition objects into a Matrix of numbers.
+"""
+function designmatrix(comps)
+    reduce(hcat, components.(comps))'
+end
+
+"""
+    compvarmatrix(comps)
+
+Returns the compositional variation array, from definition 4.3 of Aitchson - The Statistical Analysis of Compositional Data.
+"""
+function compvarmatrix(comps)
+    X = designmatrix(comps)
+    N, D = size(X)
+    cva = zeros(D, D)
+
+    for i in 1:D
+        for j in i+1:D
+            log_ratios = log.(X[:, i] ./ X[:, j])
+            cva[j, i] = sum(log_ratios) / N
+            cva[i, j] = sum((log_ratios .- cva[j, i]).^2 ) / N
```

Comment:
Please use `mean` and `var` from Statistics now that you have added it as a dependency.